fvm/src/syscalls/context.rs

```diff
@@ -81,6 +83,21 @@ impl Memory {
         .context("failed to parse cid")
     }
 
+    pub fn write_cid(&mut self, k: &Cid, offset: u32, len: u32) -> Result<u32> {
+        let out = self.try_slice_mut(offset, len)?;
+
+        let mut buf = Cursor::new([0u8; MAX_CID_LEN]);
```

Comment:
Writing then copying is likely no slower than pre-computing the size (as we were doing before). However, this is much less likely to have a bug.